src/Solocraft.cpp

```diff
@@ -328,11 +332,24 @@ class SolocraftAnnounce : public PlayerScript
             //Remove database entry as the player has logged out
             CharacterDatabase.Execute("DELETE FROM `custom_solocraft_character_stats` WHERE `GUID`={}", player->GetGUID().GetCounter());
         }
+        playerInInstanceMap.erase(player->GetGUID());
+    }
+
+    void OnMapChanged(Player* player) override
+    {
+        if (player->GetMap()->IsDungeon() || player->GetMap()->IsRaid())
+        {
+            playerInInstanceMap[player->GetGUID()] = true;
+        }
+        else
+        {
+            playerInInstanceMap[player->GetGUID()] = false;
+        }
```

Comment:
you could write the following code instead of if else
```c++
const bool isPlayerInInstanceMap = player->GetMap()->IsDungeon() || player->GetMap()->IsRaid();
playerInInstanceMap[player->GetGUID()] = isPlayerInInstanceMap;
```